crates/controller/src/hud/minimap/fill.rs

```diff
@@ -6,22 +9,56 @@ use de_core::{
 use de_map::size::MapBounds;
 use de_objects::SolidObjects;
 use de_terrain::TerrainCollider;
-use de_types::projection::ToFlat;
+use de_types::{objects::*, projection::ToFlat};
 use parry2d::{
     bounding_volume::Aabb,
     math::Point,
     query::{Ray, RayCast},
 };
 
-use super::draw::DrawingParam;
-use crate::ray::ScreenRay;
-
 const TERRAIN_COLOR: Color = Color::rgb(0.61, 0.46, 0.32);
 const PLAYER_COLOR: Color = Color::rgb(0.1, 0.1, 0.9);
-const ENEMY_COLOR: Color = Color::rgb(0.9, 0.1, 0.1);
+const ENEMY_COLORS: [Color; 4] = [
+    Color::rgb(0.1, 0.9, 0.1),
+    Color::rgb(0.9, 0.1, 0.1),
+    Color::rgb(0.9, 0.9, 0.1),
+    Color::rgb(0.1, 0.9, 0.9),
+];
 const MIN_ENTITY_SIZE: Vec2 = Vec2::splat(0.02);
 const CAMERA_COLOR: Color = Color::rgb(0.9, 0.9, 0.9);
 
+#[derive(Resource, Debug)]
+struct PlayerColors {
+    player: (Color, AHashMap<ObjectType, Color>),
+    enemies: [(Color, AHashMap<ObjectType, Color>); 4],
```

Comment:
Please replace the hard-coded 4 with `Player::MAX_PLAYERS` so that the code does not break when max players is changed.